src/NaiveCentroid.cc

```diff
@@ -26,18 +26,22 @@
 
 
 namespace lsst { namespace meas { namespace base {
-
 namespace {
+FlagDefinitionList flagDefinitions;
+} // end anonymous
+
+FlagDefinition const NaiveCentroidAlgorithm::FAILURE = flagDefinitions.addFailureFlag();
+FlagDefinition const NaiveCentroidAlgorithm::NO_COUNTS = flagDefinitions.add("flag_noCounts", "Object to be centroided has no counts");
+FlagDefinition const NaiveCentroidAlgorithm::EDGE = flagDefinitions.add("flag_edge", "Object too close to edge");
 
-std::array<FlagDefinition,NaiveCentroidAlgorithm::N_FLAGS> const & getFlagDefinitions() {
-    static std::array<FlagDefinition,NaiveCentroidAlgorithm::N_FLAGS> const flagDefs = {{
-        {"flag", "general failure flag, set if anything went wrong"},
-        {"flag_noCounts", "Object to be centroided has no counts"},
-        {"flag_edge", "Object too close to edge"}
-    }};
-    return flagDefs;
+FlagDefinitionList const & NaiveCentroidAlgorithm::getFlagDefinitions() {
+    return flagDefinitions;
 }
 
+
```

Comment:
Not sure why you have an empty namespace here.  There is another example of this below.